### PR TITLE
feat(core): specify config as interfaces

### PR DESCRIPTION
**Pull request by cal-smith**
- add interfaces for each chart types specific config
- add interface for the data format
- clean up unused variables

### CODE REVIEW

**Comment by theiliad** [comment]
@cal-smith Code is fine, demos are broken:
image

**Comment by theiliad** [comment]
Also, when resizing now, labels move

**Comment by cal-smith** [comment]
Interesting! I'll take a closer look

File: packages/core/src/base-chart.ts [hunk 1]
```diff
@@ -49,9 +49,9 @@ export class BaseChart {
 
 	constructor(holder: Element, configs: ChartConfig<BaseChartOptions>) {
 		this.id = `chart-${BaseChart.chartCount++}`;
-
+		console.log(this.options);
```
Comment (theiliad) [inline packages/core/src/base-chart.ts]: probably just missed this

Comment (cal-smith) [reply packages/core/src/base-chart.ts]: yep

**Comment by cal-smith** [comment]
Nothing breaking, it just codifies the current API

**Comment by theiliad** [comment]
Tooltips are broken now:
image

As well as border-opacity:
image

image

image

**Comment by cal-smith** [comment]
My bad, `merge` wasn't checking for falsey values when deciding what to merge ... fixed 👍

File: packages/core/src/line-chart.ts [hunk 2]
```diff
@@ -17,6 +17,15 @@ export class LineChart extends ScatterChart {
 	constructor(holder: Element, configs: ChartConfig<LineChartOptions>) {
 		super(holder, configs);
 
+		console.log(configs.options, Configuration.options.LINE);
```
Comment (theiliad) [inline packages/core/src/line-chart.ts]: 🙂

Comment (cal-smith) [reply packages/core/src/line-chart.ts]: I'll get it right sooner or later 😬

**Pull request merged**